hal/ia32/string.c

```diff
@@ -19,21 +19,22 @@
 
 void hal_memcpy(void *dst, const void *src, size_t l)
 {
-	__asm__ volatile
-	(" \
-		cld; \
-		movl %0, %%ecx; \
-		movl %%ecx, %%edx; \
-		andl $3, %%edx; \
-		shrl $2, %%ecx; \
-		movl %1, %%edi; \
-		movl %2, %%esi; \
-		rep; movsl; \
-		movl %%edx, %%ecx; \
-		rep; movsb"
+	/* clang-format off */
+	__asm__ volatile (
+		"cld\n\t"
+		"movl %0, %%ecx\n\t"
+		"movl %%ecx, %%edx\n\t"
+		"andl $3, %%edx\n\t"
```

Comment:
Shorter
```suggestion
		"andb $3, %%dl\n\t"
```

Reply:
Wrong suggestion.
For example:
```
movl $0x100, %%ecx
movl %%ecx, %%edx
andb $3, %%dl
```
 `edx` register will have a value of `0x100`  instead `0`.

Reply:
I'm pretty sure it is an invalid change. The reason is the same as what @kemonats mentioned in a comment above.

Reply:
Ack. The same. I marked conversation as resolved, wrong suggestion not be used.